weld/resources/dictmerger_global.ll

```diff
@@ -0,0 +1,67 @@
+; Template for a dictmerger and its helper functions.
+;
+; Parameters:
+; - NAME: name to give generated type, without % or @ prefix
+; - KEY: LLVM type of key (e.g. i32 or %MyStruct)
+; - VALUE: LLVM type of value (e.g. i32 or %MyStruct)
+; - KV_STRUCT: name of struct holding {KEY, VALUE} (should be generated outside)
+; - KV_VEC: name of vector of KV_STRUCTs (should be generated outside)
+; - KV_VEC_PREFIX: prefix for helper functions of KV_VEC
+; - OP: binary commutative merge operation (example: add or fadd)
+
+%$NAME.bld = type %$NAME* ; the dictmerger is a pointer to the corresponding dictionary
+
+; Initialize and return a new dictionary with the given initial capacity.
+; The capacity must be a power of 2.
+define %$NAME.bld @$NAME.bld.new(i64 %capacity) {
+  %structSizePtr = getelementptr %$NAME, %$NAME* null, i32 1
+  %structSize = ptrtoint %$NAME* %structSizePtr to i64
+  %rawPtr = call i8* @new_merger(i64 %structSize, i32 1)
+  %newDict = call %$NAME @$NAME.new(i64 %capacity)
+  %bldPtr = bitcast i8* %rawPtr to %$NAME*
+  store %$NAME %newDict, %$NAME* %bldPtr, align 1
+  ret %$NAME.bld %bldPtr
+}
+
+; Append a value into a builder, growing its space if needed.
+define %$NAME.bld @$NAME.bld.merge(%$NAME.bld %bldPtr, %$KV_STRUCT %keyValue, i32 %workerId) {
```

Comment:
Are you suggesting refactoring the linear probing logic to some sort of `dictmerger_prelude`?